server/src/test/java/org/opensearch/search/aggregations/startree/BoolStarTreeFilterProviderTests.java

```diff
@@ -0,0 +1,894 @@
+ * compatible open source license.
+ */
+
+package org.opensearch.search.aggregations.startree;
+
+import org.apache.lucene.util.BytesRef;
+import org.opensearch.index.compositeindex.datacube.Metric;
+import org.opensearch.index.compositeindex.datacube.MetricStat;
+import org.opensearch.index.compositeindex.datacube.OrdinalDimension;
+import org.opensearch.index.compositeindex.datacube.startree.StarTreeField;
+import org.opensearch.index.compositeindex.datacube.startree.StarTreeFieldConfiguration;
+import org.opensearch.index.mapper.CompositeDataCubeFieldType;
+import org.opensearch.index.mapper.KeywordFieldMapper;
+import org.opensearch.index.mapper.MapperService;
+import org.opensearch.index.mapper.NumberFieldMapper;
+import org.opensearch.index.mapper.StarTreeMapper;
+import org.opensearch.index.mapper.WildcardFieldMapper;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.RangeQueryBuilder;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.index.query.TermsQueryBuilder;
+import org.opensearch.index.query.WildcardQueryBuilder;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.startree.filter.DimensionFilter;
+import org.opensearch.search.startree.filter.ExactMatchDimFilter;
+import org.opensearch.search.startree.filter.RangeMatchDimFilter;
+import org.opensearch.search.startree.filter.StarTreeFilter;
+import org.opensearch.search.startree.filter.provider.StarTreeFilterProvider;
+import org.opensearch.test.OpenSearchTestCase;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BoolStarTreeFilterProviderTests extends OpenSearchTestCase {
+    private SearchContext searchContext;
+    private MapperService mapperService;
+    private CompositeDataCubeFieldType compositeFieldType;
+
+    @Before
+    public void setup() {
+        // Setup common test dependencies
+        searchContext = mock(SearchContext.class);
+        mapperService = mock(MapperService.class);
+        when(searchContext.mapperService()).thenReturn(mapperService);
+
+        // Setup field types
+        KeywordFieldMapper.KeywordFieldType methodType = new KeywordFieldMapper.KeywordFieldType("method");
+        NumberFieldMapper.NumberFieldType statusType = new NumberFieldMapper.NumberFieldType(
+            "status",
+            NumberFieldMapper.NumberType.INTEGER
+        );
+        NumberFieldMapper.NumberFieldType portType = new NumberFieldMapper.NumberFieldType("port", NumberFieldMapper.NumberType.INTEGER);
+        KeywordFieldMapper.KeywordFieldType zoneType = new KeywordFieldMapper.KeywordFieldType("zone");
+        NumberFieldMapper.NumberFieldType responseTimeType = new NumberFieldMapper.NumberFieldType(
```

Comment:
Can we maybe add one of the decimal types to one of the fields [ float / double ]

Reply:
Added